src/stdlib/RandomMeasures.jl

```diff
@@ -118,29 +118,26 @@ struct DirichletProcess{T<:Real} <: AbstractRandomProbabilityMeasure
     α::T
 end
 
-_rand(rng::AbstractRNG, d::StickBreakingProcess{DirichletProcess{T}}) where {T<:Real} = rand(rng, Beta(one(T), d.rpm.α))
-
-function _rand(rng::AbstractRNG, d::SizeBiasedSamplingProcess{DirichletProcess{T}}) where {T<:Real}
-    return d.surplus*rand(rng, Beta(one(T), d.rpm.α))
-end
-
-function _logpdf(d::StickBreakingProcess{DirichletProcess{T}}, x::T) where {T<:Real}
-    return logpdf(Beta(one(T), d.rpm.α), x)
+function distribution(d::StickBreakingProcess{<:DirichletProcess})
+    α = d.rpm.α
+    return Beta(one(α), α)
 end
 
-function _logpdf(d::SizeBiasedSamplingProcess{DirichletProcess{T}}, x::T) where {T<:Real}
-    return logpdf(Beta(one(T), d.rpm.α), x/d.surplus)
+function distribution(d::SizeBiasedSamplingProcess{<:DirichletProcess})
+    α = d.rpm.α
+    return LocationScale(zero(α), d.surplus, Beta(one(α), α))
 end
 
-function _logpdf_table(d::DirichletProcess{T}, m::AbstractVector{Int}) where {T<:Real}
+function _logpdf_table(d::DirichletProcess, m::AbstractVector{Int})
     # compute the sum of all cluster counts
     sum_m = sum(m)
 
     # shortcut if all cluster counts are zero
+    dα = d.α
+    T = typeof(dα)
```

Comment:
I wonder if this makes any performance difference to the original code. I would assume the original implementation should be easier to optimise for the compiler. What do you think?

Reply:
I think the compiler is smart enough to compile it away. In the example
```julia
struct A{T}
    a::T
end

f(a::A) = typeof(a.a)
g(::A{T}) where T = T
```
`f` and `g` seem to produce the same native assembly instructions, as
```julia
@code_native f(A(rand(3)))
@code_native g(A(rand(3)))
```
shows.